### PR TITLE
Read team owners from teams.yml

### PR DESCRIPTION
**Pull request by clux**
This deprecates `config.teams` in favour of `config.owners` which is read by `Config::read` from an external `teams.yml`. The spec for this new file is encoded in `teams.rs`.

The plan is to generate the `teams.yml` within babylon from another repository, and validate in shipcat that each service has a metadata.team matching one of our squads. This is done gradually in shipcat to allow phasing out the old `config.teams` references.

### CODE REVIEW

**Comment by gleaver** [comment]
CODEOWNERS and vault policies

File: shipcat_definitions/src/structs/metadata.rs [hunk 1]
```diff
@@ -167,10 +176,29 @@ impl Metadata {
...
+                    // fallback legacy
+                    let ts = teams.to_vec().into_iter().map(|t| t.name).collect::<Vec<_>>();
+                    if !ts.contains(&self.team) {
+                        bail!("Illegal team name {} not found in the config", self.team);
+                    }
+                }
+            },
+            ServiceOwnership::Squads => {
+                if !owners.squads.contains_key(&self.team) {
+                    bail!("Team name {} does not match a tribe in teams.yml", self.team);
```
Comment (gleaver) [inline shipcat_definitions/src/structs/metadata.rs]: tribe != squad

File: shipcat_filebacked/src/manifest.rs [hunk 2]
```diff
@@ -216,17 +217,49 @@ impl ManifestSource {
...
+            // Deprecated dual mode
+            ServiceOwnership::SquadsOrLegacyTeam => {
+                if let Some(t) = conf.teams.iter().find(|t| t.name == md.team) {
+                    if md.support.is_none() {
+                        md.support = t.support.clone();
+                    }
+                    if md.notifications.is_none() {
+                        md.notifications = t.notifications.clone();
+                    }
+                } else if let Some(s) = conf.owners.squads.get(&md.team) {
```
Comment (gleaver) [inline shipcat_filebacked/src/manifest.rs]: Ordering is the wrong way around, squads > teams

**Comment by clux** [comment]
This should work fine now. Output parameters `squad` and `tribe` now present on the manifest `metadata` struct when `teams.yml` has a corresponding squad entry for `metadata.team`.

This allows us to start propagating accurate squad/tribe labels to charts.

**Pull request merged**